modules/contracts/apps/payment/src/handlers/execute.rs

```diff
@@ -117,7 +126,7 @@ pub fn tip(
     }
 
     // Tip history
-    update_tipper_history(deps.storage, &info.sender, total_amount)?;
+    update_tipper_history(deps.storage, &info.sender, &desired_asset, total_amount)?;
```

Comment:
The logic for this function is incomplete imo. 

the function should accept `Vec<AnsAsset>` where 

1. Any asset that could be swapped to the desired asset is swapped and added to the desired asset's amount
2. Any assets that could NOT be swapped are added to the `Vec`. That way even payments that weren't swappable will still show up. 

Maybe a snapshot map could also be useful for this to simplify the FE integration (i.e. show different state over different time periods without needing an Indexer)

Reply:
>     2. Any assets that could NOT be swapped are added to the `Vec`. That way even payments that weren't swappable will still show up.

Hm, do we want to have some validation of that asset? Like query ans if that asset at least exist? Not sure how annoying spam assets in cosmos yet

Reply:
Ah but the assets are reversed queried so if the asset isn't in ANS is won't work anyway